unit-tests/src/test/scala/java/util/jar/JarEntrySuite.scala

```diff
@@ -0,0 +1,104 @@
+package java.util.jar
+
+// Ported from Apache Harmony
+
+import java.nio.file.{Files, Paths}
+import JarBytes._
+
+object JarEntrySuite extends tests.Suite {
+
+  test("Constructor(JarEntry)") {
+    val jarFile     = getJarFile()
+    val newJarEntry = new JarEntry(jarFile.getJarEntry(entryName))
+    assert(newJarEntry != null)
+    jarFile.close()
+  }
+
+  test("Constructor(ZipEntry)") {
+    val jarFile = getJarFile()
+    assert(jarFile != null)
+    val zipEntry = jarFile.getEntry(entryName)
+    assert(zipEntry != null)
+    val jarEntry = new JarEntry(zipEntry)
+    assert(jarEntry != null)
+    assert(jarEntry.getName() == entryName)
+    assert(jarEntry.getSize() == 311)
+    jarFile.close()
+  }
+
+  test("getAttributes()") {
+    val attrJar       = getAttJarFile()
+    val attrsJarEntry = attrJar.getJarEntry(attEntryName)
+    assert(attrsJarEntry.getAttributes() != null)
+
+    val noAttrsJarEntry = attrJar.getJarEntry(attEntryName2)
+    assert(noAttrsJarEntry.getAttributes() == null)
+    attrJar.close()
+  }
+
+  // TODO: Uncomment once we support signed jars
```

Comment:
We have `testFails("...", issue = 12345) { ... }` to predicate tests that are expected to fail with link to an issue. Lets use that instead for all of the TODO tests.